community/cloud-foundation/templates/ip_reservation/ip_reservation.py.schema

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+info:
+  title: IP Reservation
+  author: Sourced Group Inc.
+  description: Reservers internal, external, or global IP address.
+
+required:
+  - ips
+
+properties:
+  ips:
+    type: array
+    description: An array of IPs to create.
+    items:
+      type: object
+      required:
+        - name
+        - ipType
+      properties:
+        name:
+          type: string
+          description: |
+            Name of the reserved IP; unique
+            within the context of the project.
+        ipType:
+          type: string
+          enum:
+            - global
```

Comment:
Can we use 'GLOBAL', 'REGIONAL', 'INTERNAL' so no need for additional translation, plus no diversion from the API?

Reply:
The API currently supports "INTERNAL" and "EXTERNAL". Addresses are represented by a compute.v1.address where as global addresses are compute.v1.globalAddress. If this template is to handle both resources, some type of mapping will need to exist or we can add a property field to specify if the address is "GLOBAL", that way we can maintain the `ipType` to just "INTERNAL/EXTERNAL" to match with the API. 

Any preference or suggestions?

Reply:
I'm okey with the current implementation, just make it all caps so we don't have an internal > INTERNAL translation.